site/plugins/site/helpers/html.php

```diff
@@ -119,6 +122,46 @@ function version(string $version, string $format = '%s'): string
 	);
 }
 
+/**
+ * Creates a privacy-friendly video embed via iframe for YouTube
+ */
+function video(
+	string $url,
+	File $cover,
+	array $options = [],
+	array $attr = [],
+): string|null {
+	if (Str::contains($url, 'vimeo', true) === true) {
+		throw new Exception('Vimeo support is not implemented');
+	}
+
+	// delegate local video support to the core implementation
+	if (Str::contains($url, 'youtu', true) === false) {
+		return Html::video($url, $options, $attr);
+	}
+
+	// normalize video URL for the fallback link
+	$url = str_replace('youtu.be/', 'www.youtube.com/watch?v=', $url);
+	$url = str_replace('www.youtube-nocookie.com', 'www.youtube.com', $url);
+
+	// but always use privacy mode for embedding
+	$privacyUrl = str_replace('www.youtube.com', 'www.youtube-nocookie.com', $url);
+
+	// get the iframe string from the core
+	$iframe = Html::youtube($privacyUrl, [
+		...$options['youtube'] ?? [],
+		'autoplay'       => 1,
+		'modestbranding' => 1,
+		'showinfo'       => 0,
+		'rel'            => 0,
+	], [
+		...$attr,
+		'referrerpolicy' => 'no-referrer'
+	]);
+
+	return Tpl::load(dirname(__DIR__) . '/snippets/video.php', compact('attr', 'iframe', 'url', 'cover'));
```

Comment:
```suggestion
	return  snippet('video', compact('attr', 'iframe', 'url', 'cover'), true);
```

`Suggestion`: I would not start splitting up our snippets between `site/snippets/` and plugins. That will make it just harder later on to find the code. If we place it with the regular snippets, we can also use the `snippet()` helper here.